src/pycram/process_modules/icub_process_modules.py

```diff
@@ -0,0 +1,687 @@
+from ..world_concepts.world_object import Object
+from ..datastructures.pose import Pose
+from ..datastructures.enums import JointType, ObjectType, Arms, ExecutionType, GripperState
+from ..external_interfaces import giskard
+from ..external_interfaces.robokudo import *
+import yarp
+
+from pycram.yarp_utils.yarp_networking import NO_ACK_VOCAB, ACK_VOCAB, open_rpc_client_port, open_buffered_bottle_port, \
+    init_yarp_network
+
+def update_hand(hand_values,ctp_port):
+    yarp_bottle_msg: yarp.Bottle = yarp.Bottle()
+    yarp_bottle_reply: yarp.Bottle = yarp.Bottle()
+    yarp_bottle_msg.addVocab32('c', 't', 'p', 'q')
+    yarp_bottle_msg.addVocab32('t', 'i', 'm', 'e')
+    yarp_bottle_msg.addFloat32(1.5)
+    yarp_bottle_msg.addVocab32('o', 'f', 'f')
+    yarp_bottle_msg.addInt32(7)
+    yarp_bottle_msg.addVocab32('p', 'o', 's')
+    target_loc: yarp.Bottle = yarp_bottle_msg.addList()
+    for i in hand_values:
+        target_loc.addFloat32(i)
+
+    print("command Ready to send to iCub  part tcp")
+
+    ctp_port.write(yarp_bottle_msg, yarp_bottle_reply)
+    reply_vocab = yarp_bottle_reply.get(0).asVocab32()
+
+    if reply_vocab == NO_ACK_VOCAB:
+        print("NO_ACK")
+        return False
+    elif reply_vocab == ACK_VOCAB:
+        print("ACK")
+        return True
+    else:
+        print("another reply")
+        return False
+
+def update_part(state_port,ctp_port, joint_to_change_idx, joints_to_change_pos):
+    if len(joint_to_change_idx):
+        part_state: yarp.Bottle = state_port.read(shouldWait=True)
+        part_new_states = []
+        for i in range(part_state.size()):
+            print(i, "   ", part_state.get(i).asFloat32())
+            part_new_states.append(part_state.get(i).asFloat32())
+
+        for i in range(len(joint_to_change_idx)):
+            part_new_states[joint_to_change_idx[i]] = math.degrees(joints_to_change_pos[i])
+
+        yarp_bottle_msg: yarp.Bottle = yarp.Bottle()
+        yarp_bottle_reply: yarp.Bottle = yarp.Bottle()
+        yarp_bottle_msg.addVocab32('c', 't', 'p', 'q')
+        yarp_bottle_msg.addVocab32('t', 'i', 'm', 'e')
+        yarp_bottle_msg.addFloat32(1.5)
+        yarp_bottle_msg.addVocab32('o', 'f', 'f')
+        yarp_bottle_msg.addInt32(0)
+        yarp_bottle_msg.addVocab32('p', 'o', 's')
+        target_loc: yarp.Bottle = yarp_bottle_msg.addList()
+        for i in part_new_states:
+            target_loc.addFloat32(i)
+
+        print("command Ready to send to iCub  part tcp")
+
+        ctp_port.write(yarp_bottle_msg, yarp_bottle_reply)
+        reply_vocab = yarp_bottle_reply.get(0).asVocab32()
+
+        if reply_vocab == NO_ACK_VOCAB:
+            print("NO_ACK")
+            return False
+        elif reply_vocab == ACK_VOCAB:
+            print("ACK")
+            return True
+        else:
+            print("another reply")
+            return False
+
+HAND_CLOSED = [60.0,60.0,0,85.0,20,85,20,85,85]
+HAND_OPENED = [0   ,0   ,0,0   ,0 ,0 ,0 ,0 ,0]
```

Comment:
I guess these are joint values for the hand, they are already in the robot_description use these insetad and remove the global variable

Reply:
not all the joint are directly actuated 1:1. 
In the robot description I have the values for every joint for close and open. However the control joints are different and also different values. 
So this list is the actuated control values. which is only used for the real robot 
What do you think ?

Reply:
Ah ok, then you cloud put them in the move_gripper process module of the real robot. This way it should be clear where they are used.